flink-connectors/flink-connector-aws-kinesis-firehose/src/test/java/org/apache/flink/connector/firehose/sink/KinesisFirehoseSinkITCase.java

```diff
@@ -78,42 +76,36 @@
     @Before
     public void setup() throws Exception {
         System.setProperty(SdkSystemSetting.CBOR_ENABLED.property(), "false");
-        httpClient = AWSServicesTestUtils.createHttpClient(mockFirehoseContainer.getEndpoint());
-        s3AsyncClient = createS3Client(mockFirehoseContainer.getEndpoint(), httpClient);
-        firehoseAsyncClient = getFirehoseClient(mockFirehoseContainer.getEndpoint(), httpClient);
-        iamAsyncClient = createIamClient(mockFirehoseContainer.getEndpoint(), httpClient);
+        s3HttpClient = createHttpClient(mockFirehoseContainer.getEndpoint());
+        firehoseHttpClient = createHttpClient(mockFirehoseContainer.getEndpoint());
+        iamHttpClient = createHttpClient(mockFirehoseContainer.getEndpoint());
+        s3AsyncClient = createS3Client(mockFirehoseContainer.getEndpoint(), s3HttpClient);
+        firehoseAsyncClient =
+                getFirehoseClient(mockFirehoseContainer.getEndpoint(), firehoseHttpClient);
+        iamAsyncClient = createIamClient(mockFirehoseContainer.getEndpoint(), iamHttpClient);
+        env = StreamExecutionEnvironment.getExecutionEnvironment();
```

Comment:
Why did you split out to separate HTTP clients?

Reply:
So I can keep a reference to the HTTP clients to close

Reply:
This was already done though. Before the AWS SDK clients were sharing a single HTTP client. I wondered if there is a reason we needed to use 3 instead of 1?

Reply:
As discussed, this appears to address a classloader flaky test issue

Reply:
Can you elaborate?

Reply:
@CrynetLogistics reported that flaky classloader test was reproduced locally and this fixes it, can you please elaborate here as per our discussion?

Reply:
Sharing a single http client between the 3 service clients causes the ci and local to fail intermittently. Frequency approx. 1 in 6. This is reproducible. Changing them to use separate ones fixes this issue and I have ran it over 380 times on my local without failure. The issue relates to `classloaders` being closed prior to usage of the clients.

Reply:
Can you be more specific? This doesn't really explain what's causing the issue. What exect class-loader is being closed and how it relates to the AWS client? Can you point to a specific place in the code that causes the issue?

Reply:
There is something sketchy going with thread locals and XML parser service loading. This can signal some deeper issues, so we should make sure why exactly this is happening, even though we have a "workaround" that works for some reason.

```
Feb 10 10:57:30 Caused by: java.lang.IllegalStateException: Trying to access closed classloader. Please check if you store classloaders directly or indirectly in static fields. If the stacktrace suggests that the leak occurs in a third party library and cannot be fixed immediately, you can disable this check with the configuration 'classloader.check-leaked-classloader'.
Feb 10 10:57:30 	at org.apache.flink.runtime.execution.librarycache.FlinkUserCodeClassLoaders$SafetyNetWrapperClassLoader.ensureInner(FlinkUserCodeClassLoaders.java:164)
Feb 10 10:57:30 	at org.apache.flink.runtime.execution.librarycache.FlinkUserCodeClassLoaders$SafetyNetWrapperClassLoader.getResources(FlinkUserCodeClassLoaders.java:188)
Feb 10 10:57:30 	at java.util.ServiceLoader$LazyIterator.hasNextService(ServiceLoader.java:348)
Feb 10 10:57:30 	at java.util.ServiceLoader$LazyIterator.hasNext(ServiceLoader.java:393)
Feb 10 10:57:30 	at java.util.ServiceLoader$1.hasNext(ServiceLoader.java:474)
Feb 10 10:57:30 	at javax.xml.stream.FactoryFinder$1.run(FactoryFinder.java:352)
Feb 10 10:57:30 	at java.security.AccessController.doPrivileged(Native Method)
Feb 10 10:57:30 	at javax.xml.stream.FactoryFinder.findServiceProvider(FactoryFinder.java:341)
Feb 10 10:57:30 	at javax.xml.stream.FactoryFinder.find(FactoryFinder.java:313)
Feb 10 10:57:30 	at javax.xml.stream.FactoryFinder.find(FactoryFinder.java:227)
Feb 10 10:57:30 	at javax.xml.stream.XMLInputFactory.newInstance(XMLInputFactory.java:154)
Feb 10 10:57:30 	at software.amazon.awssdk.protocols.query.unmarshall.XmlDomParser.createXmlInputFactory(XmlDomParser.java:124)
Feb 10 10:57:30 	at java.lang.ThreadLocal$SuppliedThreadLocal.initialValue(ThreadLocal.java:284)
Feb 10 10:57:30 	at java.lang.ThreadLocal.setInitialValue(ThreadLocal.java:180)
Feb 10 10:57:30 	at java.lang.ThreadLocal.get(ThreadLocal.java:170)
Feb 10 10:57:30 	at software.amazon.awssdk.protocols.query.unmarshall.XmlDomParser.parse(XmlDomParser.java:44)
Feb 10 10:57:30 	at software.amazon.awssdk.protocols.xml.internal.unmarshall.XmlResponseParserUtils.parse(XmlResponseParserUtils.java:56)
Feb 10 10:57:30 	at software.amazon.awssdk.protocols.xml.internal.unmarshall.AwsXmlPredicatedResponseHandler.parseResponse(AwsXmlPredicatedResponseHandler.java:114)
Feb 10 10:57:30 	at software.amazon.awssdk.protocols.xml.internal.unmarshall.AwsXmlPredicatedResponseHandler.handleResponse(AwsXmlPredicatedResponseHandler.java:95)
Feb 10 10:57:30 	at software.amazon.awssdk.protocols.xml.internal.unmarshall.AwsXmlPredicatedResponseHandler.handle(AwsXmlPredicatedResponseHandler.java:84)
Feb 10 10:57:30 	at software.amazon.awssdk.protocols.xml.internal.unmarshall.AwsXmlPredicatedResponseHandler.handle(AwsXmlPredicatedResponseHandler.java:42)
Feb 10 10:57:30 	at software.amazon.awssdk.core.internal.handler.BaseClientHandler.lambda$successTransformationResponseHandler$6(BaseClientHandler.java:232)
```

Reply:
What is strange is why these clients are using the Flink user classloader to start with. This test is running an Embedded Mini Cluster, I assume that Flink is reusing the classloader from the unit test (a guess). If this is the case it could be a race condition between the Flink job finishing and and clients closing. 

We also discussed using a sync client, as the stack trace we saw previously came from one of the AWS SDK thread pools. But I would like to understand why this happens.

Reply:
Any ideas on what might be the root cause here? This is something I spent Wednesday, Thursday and Friday looking at last week, with unfortunately no breakthrough. However, I did find that separate http clients resolved the issue.

Would you be able to assist? I believe the root cause is not in this codebase but in one of the sdk clients.

Reply:
@dmvk I spent the past hour trying to reproduce it... I think I'm close, I have pushed what I have so far on to [FLINK-25846-commit-f2e288b](https://github.com/CrynetLogistics/flink/tree/FLINK-25846-commit-f2e288b). It hangs locally rather than failing with `classloader`.

The main issue is because we rebased this branch on master quite a few times now, something may have been introduced to change this behaviour. Anyhow it should be at or near this commit (f2e288b) in this PR's history.

Reply:
@dmvk Just failed in this [PR](https://github.com/apache/flink/pull/18686) and it's corresponding [CI build](https://dev.azure.com/apache-flink/apache-flink/_build/results?buildId=31136&view=results).

Since this is something that is currently impacting `master` and it is apparent that this PR fixes it, would you be happy to merge this and raise a follow up ticket for investigation?

Reply:
But I know this would fix the problem, since I was able to reproduce it locally, and the final commit resolves the issue. Surely merging this is a better solution.

Reply:
It fixes the problem and you don't know why, what kind of fix is that?

Reply:
@dmvk Do I rebase on top of the commit you just pushed removing the `@Ignore` and merge this commit once the CI passes?